pkg/build/api/v1/types.go

```diff
@@ -395,6 +399,36 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification. The hook
+// executes a command in a temporary container running the build output image,
+// immediately after the last layer of the image is commited and before the
+// image is pushed to a registry. If none of the fields are specified, the hook
+// is not executed.
+type BuildPostCommitSpec struct {
+	// Command is the command to run. It may not be specified with Script.
+	// The Docker image's entrypoint is used if this is not provided.
+	// Use Command when you need to control what is the image entrypoint.
+	// This might be needed if the image doesn't have "/bin/sh", or if you
+	// cannot afford using a shell. In all other cases, using Script will be
+	// more convenient.
+	Command []string `json:"command,omitempty" description:"command to be executed in a container running the build output image replacing the image's entrypoint"`
+	// Args is a list of arguments that are provided to either Command,
+	// Script or the Docker image's default entrypoint. In all cases, the
+	// arguments are placed immediately after the command to be run.
+	// For Script, "/bin/sh" is prepended to the argument list, so that it
+	// will become $0 in the script's context, while the arguments in Args
+	// will become $1, $2, etc.
+	// E.g., given `{Script: "echo $0 $1 $2", Args: {"foo", "bar"}}` the
+	// post commit hook will print to stdout "/bin/sh foo bar".
+	Args []string `json:"args,omitempty" description:"arguments to command, script or the default image entrypoint"`
+	// Script is a shell script to be run with `/bin/sh -c`. It is a
+	// shortcut to specifying `{Command: {"/bin/sh", "-c"}, Args: {Script,
+	// "/bin/sh", Args...}}`.
```

Comment:
I believe the question is answered.
Reiterating: $0 will be consistently "/bin/sh", no matter if Args are used or not.

The script (or any other command) runs in the image's WORKDIR.

If the script needs to know the current directory, use `$PWD`, no need for any hacks.